src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -542,35 +549,24 @@ namespace OpenMS
-            ionseries[*ft_it + String(i) + "-36" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_h2oh2o.getMonoWeight() / charge, round_decPow);
-            ionseries[*ft_it + String(i) + "-44" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_co2.getMonoWeight() / charge, round_decPow);
-            ionseries[*ft_it + String(i) + "-46" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_hccoh.getMonoWeight() / charge, round_decPow);
-            if (sequence.toString().find("N") != std::string::npos || sequence.toString().find("Q") != std::string::npos)
-            // This hack is implemented to enable the annotation of residue specific modifications in the decoy fragments.
-            // If the function is used for generic annotation, use ion.toString() instead of sequence.toString().
-            {
-              ionseries[*ft_it + String(i) + "-45" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_ch3no.getMonoWeight() / charge, round_decPow);
-            }
-            if (sequence.toString().find("M(Oxidation)") != std::string::npos)
-            // This hack is implemented to enable the annotation of residue specific modifications in the decoy fragments.
-            // If the function is used for generic annotation, use ion.toString() instead of sequence.toString().
-            {
-              ionseries[*ft_it + String(i) + "-64" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_ch4so.getMonoWeight() / charge, round_decPow);
-            }
-            if (sequence.toString().find("S(Phospho)") != std::string::npos || sequence.toString().find("T(Phospho)") != std::string::npos)
-            // This hack is implemented to enable the annotation of residue specific modifications in the decoy fragments.
-            // If the function is used for generic annotation, use ion.toString() instead of sequence.toString().
+            if (ion[j].hasNeutralLoss())
             {
-              ionseries[*ft_it + String(i) + "-80" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_hpo3.getMonoWeight() / charge, round_decPow);
-              ionseries[*ft_it + String(i) + "-98" + "^" + String(charge)] = Math::roundDecimal(pos - neutralloss_hpo3h2o.getMonoWeight() / charge, round_decPow);
+              const std::vector<EmpiricalFormula> losses = ion[j].getLossFormulas();
+              for (std::vector<EmpiricalFormula>::const_iterator lit = losses.begin(); lit != losses.end(); ++lit)
+              {
+                if (enable_specific_losses && lit->toString() != String("H2O1") && lit->toString() != String("H3N1") && lit->toString() != String("C1H2N2") && lit->toString() != String("C1H2N1O1"))
+                {
+                  ionseries[*ft_it + String(i) + "-" + lit->toString() + "^" + String(charge)] = Math::roundDecimal(pos - lit->getMonoWeight() / charge, round_decPow);
```

Comment:
Yes, that was tricky to figure out. I added documentation to MRMIonSeries.h describing the annotation.